recipe/spin/spin_trainer.py

```diff
@@ -549,22 +549,24 @@ def _create_dataloader(self, train_dataset, val_dataset, collate_fn, train_sampl
             collate_fn=collate_fn,
             sampler=train_sampler,
         )
-
-        val_batch_size = self.config.data.val_batch_size  # Prefer config value if set
-        if val_batch_size is None:
-            val_batch_size = len(self.val_dataset)
-
-        self.val_dataloader = StatefulDataLoader(
-            dataset=self.val_dataset,
-            batch_size=val_batch_size,
-            num_workers=self.config.data.get("dataloader_num_workers", 8),
-            shuffle=False,
-            drop_last=False,
-            collate_fn=collate_fn,
-        )
-
         assert len(self.train_dataloader) >= 1, "Train dataloader is empty!"
-        assert len(self.val_dataloader) >= 1, "Validation dataloader is empty!"
+
+        if self.val_dataset is None:
```

Comment:
It seems that self.val_dataset will never be None according to the line 535?

Reply:
`create_rl_dataset` will return None is `self.config.data.val_files` is not set